src/components/Modals/UploadButtonGroup.jsx

```diff
@@ -19,22 +21,48 @@ import { useTheme } from '@mui/material/styles';
  * @param {Function} Props.setFile - The set function for handling files
  * @returns {React.JSX.Element} - The UploadButtonGroup Component
  */
+
+const dataURItoBlob = (dataURI) => {
+  // Convert base64/URLEncoded data component to raw binary data held in a string
+  let byteString;
+  if (dataURI.split(',')[0].indexOf('base64') >= 0) byteString = atob(dataURI.split(',')[1]);
+  else byteString = unescape(dataURI.split(',')[1]);
+
+  // Separate the mime component
+  const mimeString = dataURI.split(',')[0].split(':')[1].split(';')[0];
+
+  // Write the bytes of the string to a typed array
+  const ia = new Uint8Array(byteString.length);
+  for (let i = 0; i < byteString.length; i += 1) {
+    ia[i] = byteString.charCodeAt(i);
+  }
+
+  return new Blob([ia], { type: mimeString });
+};
+
 const UploadButtonGroup = ({ file, setFile }) => {
```

Comment:
While it utilizes screen width to assume what devices might have front-facing cameras, the `UploadButtonGroup` component was tailored-made for that purpose. Think you could create a new component in this file or in the Modals directory for the button group for the web cam and render it based on the new condition of whether or not there's a camera, webcam, or neither.